src/main/java/bowling/controller/BowlingController.java

```diff
@@ -24,35 +25,54 @@ public BowlingController(InputView inputView, OutputView outputView) {
     }
 
     public void run() {
-        final Player player = new Player(playerName());
-        outputView.printScoreBoard(player);
+        final Players players = players();
+        outputView.printScoreBoard(players);
 
+        playGame(players);
```

Comment:
```suggestion
        players.playGame();
```
컨트롤러가 많이 길어졌습니다. 혹시 이렇게 개선하고 이하 게임 컨트롤 부분을 `Players` 내부로 넣어보면 어떨까요?
아마도 중간 입력/출력문이 문제가 될텐데 어떤 방법이 있을지 생각해 봅시다. 😀

Reply:
저도 이런식의 구현을 생각해보긴 했었는데,
"아무래도 `BowlingGame` 같은걸 만들어서 DI를 해줘야하나?" 이런 생각도 해봤던 것 같아요.
좀 더 고민해보겠습니다.

Reply:
입력 출력을 분리하는게 어렵네요.

Reply:
템플릿 콜백 메서드 패턴을 써 보는건 어떨까요.